test/user.test.js

```diff
@@ -1770,6 +1770,293 @@ describe('User', function() {
+            User.login(
+              { email: currentEmailCredentials.email,
+                password: currentEmailCredentials.password },
+                function(err, accessToken1) {
+                  if (err) return next(err);
+                  assert(accessToken1.userId);
+                  next();
+                });
+          },
+          function secondLoginWithOriginalEmail(next) {
+            User.login({
+              email: currentEmailCredentials.email,
+              password: currentEmailCredentials.password },
+              function(err, accessToken2) {
+                if (err) return next(err);
+                assert(accessToken2.userId);
+                next();
+              });
+          },
+        ], function(err) {
+          done();
+        });
+      });
+
+      afterEach('Deleting original user', function deleteOriginalAccounts(done) {
+        User.deleteById(user.id, function(err, user) {
+          if (err) return done (err);
+          done();
+        });
+      });
+
+      it('invalidates sessions when email is changed using `updateAttribute`', function(done) {
+        user.updateAttribute('email', updatedEmailCredentials.email, function(err, userInstance) {
+          if (err) return done(err);
+          assertNoAccessTokens(done);
+        });
+      });
+
+      it('invalidates sessions when email is changed using `updateAttributes`', function(done) {
+        user.updateAttributes({ 'email': updatedEmailCredentials.email },
+        function(err, userInstance) {
+          if (err) return done(err);
+          assertNoAccessTokens(done);
+        });
+      });
+
+      it('invalidates sessions when email is changed using `replaceAttributes`', function(done) {
+        user.replaceAttributes(
+          { 'email': updatedEmailCredentials.email,
+            'password': updatedEmailCredentials.password },
+              function(err, userInstance) {
+                if (err) return done(err);
+                assertNoAccessTokens(done);
+              });
+      });
+
+      it('invalidates sessions when email is changed using `updateOrCreate`', function(done) {
+        User.updateOrCreate(
+          { id: user.id,
+            email: updatedEmailCredentials.email,
+            password: updatedEmailCredentials.password },
+            function(err, userInstance) {
+              if (err) return done(err);
+              assertNoAccessTokens(done);
+            });
+      });
+
+      it('invalidates sessions when the email is changed using `replaceById`', function(done) {
+        User.replaceById(user.id,
+          { email: updatedEmailCredentials.email,
+            password: updatedEmailCredentials.password },
+            function(err, userInstance) {
+              if (err) return done(err);
+              assertNoAccessTokens(done);
+            });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `updateAttribute`', function(done) {
```

Comment:
Since updateAttribute calls updateAttributes under the hood, I think this test is redundant and can be removed